File: 백준/Bronze/25304. 영수증/영수증.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int x = Integer.parseInt(br.readLine());
        int n = Integer.parseInt(br.readLine());
        String s;
        StringTokenizer st;
        int a;
        int b;
        int sum = 0;
        for(int i=0; i<n; i++) {
            s = br.readLine();
            st = new StringTokenizer(s, " ");
            a = Integer.parseInt(st.nextToken());
            b = Integer.parseInt(st.nextToken());
            sum += a*b;
        }
        
        if(sum == x)
            System.out.println("Yes");
        else
            System.out.println("No");
    }
}

File: 백준/Bronze/25304. 영수증/영수증.kt
fun main() = with(System.`in`.bufferedReader()) {
    val x = readLine().toInt()
    val n = readLine().toInt()
    var sum = 0
    repeat(n) {
        val (a, b) = readLine().split(" "). map { it.toInt() }
        sum += a * b
    }
    if (sum == x) {
        print("Yes")
    } else {
        print("No")
    }
}